src/app/twinclass/[twinClassId]/twinStatus/[twinStatusId]/page.tsx

```diff
@@ -61,12 +66,19 @@ export default function TwinClassPage({
           }
           setTwinStatus(data.twinStatus);
         })
-        .catch((e) => {
-          console.error("exception while fetching twin class", e);
-          toast.error("Failed to fetch twin class");
-        })
-        .finally(() => setLoading(false));
+      );
     }
+
+    if (isUndefined(twinClassCtx)) {
+      promises.push(fetchTwinClassById(twinClassId).then(setTwinClass));
+    }
+
```

Comment:
I don't think we should handle twinClass missing from context in child components. Instead such check should be in context itself, and context shouldn't render it's children if it has failed to load.